runners/core-java/src/main/java/org/apache/beam/runners/core/SplittableParDo.java

```diff
@@ -369,17 +570,17 @@ public PipelineOptions getPipelineOptions() {
         }
 
         public <T> void sideOutput(TupleTag<T> tag, T output) {
-          // TODO: I'm not sure how to implement this correctly: there's no
-          // "internals.sideOutputWindowedValue".
-          throw new UnsupportedOperationException(
-              "Side outputs not yet supported by splittable DoFn");
+          outputManager.output(
+              tag,
+              WindowedValue.of(
+                  output, element.getTimestamp(), element.getWindows(), element.getPane()));
+          noteOutput();
         }
 
         public <T> void sideOutputWithTimestamp(TupleTag<T> tag, T output, Instant timestamp) {
-          // TODO: I'm not sure how to implement this correctly: there's no
-          // "internals.sideOutputWindowedValue".
-          throw new UnsupportedOperationException(
-              "Side outputs not yet supported by splittable DoFn");
+          outputManager.output(
```

Comment:
There should be a `sideOutputWindowedValue` to match `outputWindowedValue`, for sure, on whatever we end up with.